### PR TITLE
travis: add testing

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

### CODE REVIEW

File: .travis/check-generate.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+go generate gonum.org/v1/gonum/blas
```
Comment (sbinet) [inline .travis/check-generate.sh]: not sure we need this ?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# Get deps, build, test, and ensure the code is gofmt'ed.
+# If we are building as gonum, then we have access to the coveralls api key, so we can run coverage as well.
+script:
+ - ${TRAVIS_BUILD_DIR}/.travis/check-formatting.sh
+ - go get -d -t -v ./...
+ - go build -v ./...
+ - if [[ $TRAVIS_SECURE_ENV_VARS = "true" ]]; then bash ./.travis/test-coverage.sh; fi
+ - ${TRAVIS_BUILD_DIR}/.travis/check-imports.sh
+ # This is run last since it alters the tree.
+ - ${TRAVIS_BUILD_DIR}/.travis/check-generate.sh
```
Comment (sbinet) [inline .travis.yml:36]: probably not needed ATM.

File: .travis/test-coverage.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
Comment (sbinet) [inline .travis/test-coverage.sh:1]: this is actually something that could be rewritten in Go and put as a standalone command in `gonum/tools` :)
(this script is using many UNIX commands...)

File: .travis/test-coverage.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	popd > /dev/null
+	# return our return value
+	return $retval
+}
+
+# Init acc.out
+echo "mode: set" > $ACC_OUT
+
+# Run test coverage on all directories containing go files except testlapack and testblas.
+find . -type d -not -path '*testlapack*' -and -not -path '*testblas*' | while read d; do testCover $d || exit; done
```
Comment (sbinet) [inline .travis/test-coverage.sh]: do we need this, here in this repo?

**Review by sbinet — DISMISSED** [review]
LGTM modulo some minor nits.

File: .travis/check-generate.sh (hunk 1 above)
Comment (kortschak) [reply .travis/check-generate.sh]: Yeah, probably not.

File: .travis.yml (hunk 2 above)
Comment (kortschak) [reply .travis.yml:36]: I'll leave this one here though.

File: .travis/test-coverage.sh (hunk 3 above)
Comment (kortschak) [reply .travis/test-coverage.sh:1]: File an issue for that? It can live here and be used Gonum-wide.

File: .travis/test-coverage.sh (hunk 4 above)
Comment (kortschak) [reply .travis/test-coverage.sh]: The `-not -path` expressions can go away.

**Comment by kortschak** [comment]
PTAL

File: .travis/test-coverage.sh (hunk 3 above)
Comment (sbinet) [reply .travis/test-coverage.sh:1]: done: https://github.com/gonum/tools/issues/5

**Review by sbinet — APPROVED** [review]

**Pull request merged**